src/mon/OSDMonitor.cc

```diff
@@ -1382,7 +1384,365 @@ void OSDMonitor::_check_health(
 {
   dout(20) << __func__ << dendl;
   health_check_map_t next;
-#warning write me
+  int num_osds = osdmap.get_num_osds();
+
+  // NO_OSDS
+  if (num_osds == 0) {
+    next.add("NO_OSDS", HEALTH_ERR, "no osds");
+    goto out;
+  }
+
+  // OSD_DOWN
+  // OSD_$subtree_DOWN
+  // OSD_ORPHAN
+  if (num_osds >= 0) {
```

Comment:
This if block is a little weird following the '''goto out'''' above. Especially because it's so long and covers a bunch of different tests differently than the following blocks.